app/views/whitehall_migration/documents.html.erb

```diff
@@ -1,6 +1,38 @@
-<h1>Documents</h1>
+<% content_for :back_link, render_back_link(href: whitehall_migration_path(@whitehall_migration)) %>
+<% content_for :title, "Document Imports" %>
 
-<% @documents.each do |document| %>
-  <li>ID: <%= document.id %></li>
-  <li>Status: <%= document.state %></li>
-<% end %>
+<% states = ["imported", "sync_failed", "completed"] %>
+
+<% items = @documents.map do |document_import|
+    {
+      link: {
+        text: states.include?(document_import.state) ? document_import.document.current_edition.title : document_import.id.to_s,
```

Comment:
You may want to put a method of current_title on document_import that does the whole document.current_edition.title lookup and returns nil if there is no document. You could then use the presence of document_import.current_title to decide what to show rather than needing to do this state check.